### PR TITLE
Support reading the consul token from a Kubernetes Secret

### PR DESCRIPTION
**Pull request by jeinwag**
This pull requests implements support for reading the consul token from a Kubernetes Secret, thus enabling the used of sealed secrets etc.

I had to update some dependencies to get the project to build.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ kube-consul-register watches Kubernetes events and converts information about PO
         time in seconds, what period of time will be done cleaning of inactive services (default 30m0s)
   -configmap string
         name of the ConfigMap that containes the custom configuration to use (default "default/kube-consul-register-config")
+  -secret string
```
Comment (tczekajlo) [inline README.md:14]:
```suggestion
  -consul-secret string
```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ kube-consul-register watches Kubernetes events and converts information about PO
...
+        name of the Secret that containts the consul token, key must be consul_token
```
Comment (tczekajlo) [inline README.md:15]: I assume that syntax for a secret here is `namespace/secret_name`, can you add an example in the description to keep it clear?

File: consul/consul.go [hunk 3]
```diff
@@ -44,6 +45,13 @@ func (c *Adapter) New(cfg *config.Config, podNodeName string, podIP string) *Ada
 		glog.Fatalf("bad adapter uri: ")
 	}
 
+	cfg.Consul.HttpClient = &http.Client{
```
Comment (tczekajlo) [inline consul/consul.go:48]: Why is it needed? Can't it be moved to the `default` in `switch` few lines below?

File: main.go [hunk 4]
```diff
@@ -34,6 +34,7 @@ var (
 	watchNamespace       = flag.String("watch-namespace", v1.NamespaceAll, "namespace to watch for Pods. Default is to watch all namespaces")
 	kubeconfig           = flag.String("kubeconfig", "./kubeconfig", "absolute path to the kubeconfig file")
 	configMap            = flag.String("configmap", "default/kube-consul-register-config", "name of the ConfigMap that containes the custom configuration to use")
+	secret               = flag.String("secret", "", "name of the secret containing the consul token. Key must be consul_token.")
```
Comment (tczekajlo) [inline main.go:37]:
```suggestion
	consulSecret               = flag.String("consul-secret", "", "name of the secret containing the consul token. Key must be consul_token.")
```

File: main.go [hunk 5]
```diff
@@ -95,6 +96,19 @@ func main() {
 		glog.Infof("Current configuration: Controller: %#v, Consul: %#v", cfg.Controller, cfg.Consul)
 	}
 
+	if *secret != "" {
```
Comment (tczekajlo) [inline main.go:99]:
```suggestion
	if *consulSecret != "" {
```

**Review by tczekajlo — CHANGES_REQUESTED** [review]
Thanks a lot for your PR! 🙌🏻  I've left some small comments 😄

File: consul/consul.go (hunk 3 above)
Comment (jeinwag) [reply consul/consul.go:48]: Due to a change introduced in hashicorp/consul@c6d0b4cf907bbbe10404a55aa17bff339142fc8d, the HttpClient in the config returned by DefaultConfig() is nil. I will move it within the switch block.

**Review by tczekajlo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**